numpy/__init__.pyi

```diff
@@ -488,24 +505,24 @@ class complex128(complexfloating):
 class flexible(_real_generic): ...  # type: ignore
 
 class void(flexible):
-    def __init__(self, __value: Union[int, integer, bool_, bytes, bytes_]): ...
+    def __init__(self, __value: Union[int, bool_, bytes]): ...
 
 class character(_real_generic): ...  # type: ignore
 
-class bytes_(character):
+class bytes_(character, bytes):
     @overload
     def __init__(self, __value: object = ...) -> None: ...
     @overload
     def __init__(
-        self, __value: Union[str, str_], encoding: str = ..., errors: str = ...
+        self, __value: str, encoding: str = ..., errors: str = ...
     ) -> None: ...
 
-class str_(character):
+class str_(character, str):
     @overload
     def __init__(self, __value: object = ...) -> None: ...
     @overload
     def __init__(
-        self, __value: Union[bytes, bytes_], encoding: str = ..., errors: str = ...
+        self, __value: bytes, encoding: str = ..., errors: str = ...
     ) -> None: ...
```

Comment:
For the sake of a clean history and being able to tell which types are "true" vs a compromise, it might make sense to make this change by itself, and then add the "false"-but-convenient base classes in a separate PR.

Reply:
Probably for the best, yes.

I was thinking about moving this to the maintenance pull request planned as https://github.com/numpy/numpy/pull/17172 follow up (the same one that's planning to deal with the current `generic.real` / `.imag` situation).
I'll remove the lines from here once that PR has been created.

Reply:
Ok, the `str` / `bytes` subclassing has been moved to https://github.com/numpy/numpy/pull/17214.